### PR TITLE
Retries added to the odin backend APIs

### PR DESCRIPTION
**Pull request by BalkishanNanavath**
# Odin version(x.y.z)

### CODE REVIEW

File: internal/backend/client.go [hunk 1]
```diff
@@ -27,6 +37,28 @@ func (c *clientProperties) action(entity, requestType string, body interface{})
 	return req.Make()
 }
 
+func (c clientProperties) actionWithRetry(entity, requestType string, body interface{}) request.Response {
```
Comment (brownhash) [inline internal/backend/client.go:40]: `func (c clientProperties)` should be `func (c *clientProperties)`

Comment (BalkishanNanavath) [reply internal/backend/client.go]: updated

File: internal/backend/client.go [hunk 2]
```diff
@@ -1,9 +1,19 @@
...
 	"github.com/dream11/odin/internal/config"
+	"github.com/dream11/odin/internal/ui"
 	"github.com/dream11/odin/pkg/request"
 	"github.com/dream11/odin/pkg/sse"
+	"time"
+)
+
+var logger ui.Logger
+
+const (
```
Comment (brownhash) [inline internal/backend/client.go:14]: shouldn't these be allowed to be overridden, in the actions themselves?

**Review by akshaypatidar1999 — APPROVED** [review]

File: internal/backend/client.go (hunk 2 above)
Comment (BalkishanNanavath) [reply internal/backend/client.go:15]: these constants are backend-service specific and will change for each backend-service we could have different retries but for the same backend-service why would we need different retries?

**Review by akshaypatidar1999 — APPROVED** [review]

**Review by surajgour-d11 — APPROVED** [review]

**Pull request merged**